### PR TITLE
adds e2e for variable replacement

### PR DESCRIPTION
**Pull request by guido732**
## Motivation / Description
Add a test that checks variable replacement is working correctly in the SDK

### CODE REVIEW

File: examples/rcbilling-demo/src/tests/main.test.ts [hunk 1]
```diff
@@ -77,6 +79,38 @@ test.describe("Main", () => {
     const title = page.getByText("E2E Tests for Purchases JS");
     await expect(title).toBeVisible();
   });
+  test("Can render an RC Paywall using variables", async ({
```
Comment (tonidero) [inline examples/rcbilling-demo/src/tests/main.test.ts:82]: We probably should start splitting tests into multiple files 😅 but that can come in a separate PR.

File: examples/rcbilling-demo/src/tests/main.test.ts [hunk 2]
```diff
@@ -77,6 +79,38 @@ test.describe("Main", () => {
...
+    // Gets all packages
+    const packageCards = await getAllElementsByLocator(page, PACKAGE_SELECTOR);
+
+    // Get the purchase button as a Locator
+    const purchaseButton = (
+      await getAllElementsByLocator(page, "button.purchase-button")
+    )[0];
+
+    await expect(purchaseButton).toContainText(
+      "PURCHASE FOR $1.25/1wk($5.00/mo)",
```
Comment (tonidero) [inline examples/rcbilling-demo/src/tests/main.test.ts:103]: I'm thinking... Would this break with the localization changes? We probably should hardcode the locale for the E2E tests, not sure if we're doing that... In any case, it's ok if it's passing based off `main` :)

**Review by tonidero — APPROVED** [review]
LGTM!

File: examples/rcbilling-demo/src/tests/main.test.ts (hunk 1 above)
Comment (guido732) [reply examples/rcbilling-demo/src/tests/main.test.ts:81]: agreed (eventually)

File: examples/rcbilling-demo/src/tests/main.test.ts (hunk 2 above)
Comment (guido732) [reply examples/rcbilling-demo/src/tests/main.test.ts:100]: we're not doing that, but I agree it _could_ fail, although I'm guessing the fallback language for our CI pipe "browser" is english so it should be picking that up. Either way, we should have new localized tests soon that cover these cases :)

**Pull request merged**